services/background-worker/src/Wire/BackendNotificationPusher.hs

```diff
@@ -148,12 +175,47 @@ getRemoteDomains = do
 startWorker :: RabbitMqAdminOpts -> AppT IO ()
 startWorker rabbitmqOpts = do
   env <- ask
-  liftIO . openConnectionWithRetries env.logger (demoteOpts rabbitmqOpts) $
-    RabbitMqHooks
-      { onNewChannel =
-          runAppT env . startPusher,
-        onChannelException = \_ ->
-          runAppT env $ markAsNotWorking BackendNotificationPusher,
-        onConnectionClose =
-          runAppT env $ markAsNotWorking BackendNotificationPusher
-      }
+  -- AsyncCancelled is used when our `Async ()` is `cancel`led.
+  -- This is used in the POSIX signal handlers, so we should catch it
+  -- here and clean up our processes, letting them finish if we can.
+  -- Passed into running threads so we can cancel consumers and allow
+  -- amqp to cleanly finish before we stop the service.
+  consumersRef <- newIORef mempty
+  let -- cleanup the refs when channels die
+      -- This is so we aren't trying to close consumers
+      -- that don't exist when the service is shutdown.
+      l = logger env
+      clearRefs = do
+        atomicWriteIORef consumersRef mempty
+      cleanup chan = do
+        readIORef consumersRef >>= traverse_ \(consumer, mvar) -> do
+          Log'.info l $ Log.msg (Log.val "Cancelling") . Log.field "Consumer" consumer
```

Comment:
```suggestion
          Log'.info l $ Log.msg (Log.val "Cancelling consumer") . Log.field "consumer" consumer
```

The consumer tag is just a number, probably not as useful as the domain name, which should be the key of the `Map`, maybe use `Map.traverseWithKey`.

Reply:
:+1: